src/calc_phi_ase_graybat.cc

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright 2013 Erik Zenker, Carlchristian Eckert, Marius Melzer
+ *
+ * This file is part of HASEonGPU
+ *
+ * HASEonGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * HASEonGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with HASEonGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// STL
+#include <vector>
+#include <iostream>  /* std::cout     */
+#include <algorithm> /* std::generate */
+
+// haseongpu
+#include <calc_phi_ase_mpi.hpp>
+#include <calc_phi_ase.hpp>
+#include <mesh.hpp>
+#include <logging.hpp>
+#include <progressbar.hpp>
+
+
+// GrayBat
+#include <graybat.hpp>
+#include <pattern/FullyConnected.hpp>
```

Comment:
Personal opinion:
the include should be of the form `#include <graybat/pattern/FullyConnected.hpp>`, so that it is clear from the include name itself, that it belongs to graybat.
This would require a changed include-dir in the build, I guess.